### PR TITLE
[Event Hubs] Eventhubs v1 to v5 migration guide

### PR DESCRIPTION
**Pull request by YijunXieMS**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/eventhub/azure-eventhub/migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+* `EventHubProducerClient` for sending messages. [Sync API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b6/azure.eventhub.html#azure.eventhub.EventHubProducerClient)
+and [Async API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b6/azure.eventhub.aio.html#azure.eventhub.aio.EventHubProducerClient)
+* `EventHubConsumerClient` for receiving messages. [Sync API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b6/azure.eventhub.html#azure.eventhub.EventHubConsumerClient)
+and [Async API](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-eventhub/5.0.0b6/azure.eventhub.aio.html#azure.eventhub.aio.EventHubConsumerClient)
+
+We've also merged the functionality from `EventProcessorHost` into 
+`EventHubConsumerClient`, allowing `EventHubConsumerClient` to be the single
+point of entry for receiving of any type (from single partition, all partitions, or with load balancing and checkpointing features) within Event Hubs.
+
+V5 has both sync and async APIs. Sync API is under package azure.eventhub whereas async API is under package azure.eventhub.aio.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: I think we should use terminology `module` here. When we say a package, we usually mean packages like "azure-eventhub".

So it should be:
Sync API is under **module** azure.eventhub whereas async API is under **module** azure.eventhub.aio.

File: sdk/eventhub/azure-eventhub/migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+They have the same class names under the two packages. For instance, class `EventHubConsumerClient` with sync API under package `azure.eventhub` has its 
+async counterpart under package `auzre.eventhub.aio`.
+The code samples in this migration guide use async APIs.
+
+### Client constructors
+
+| In v1                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient()`    | `EventHubProducerClient()` or `EventHubConsumerClient()` | [using credential](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhub/samples/sync_samples/client_secret_auth.py) |
```
Comment (conniey) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: One cool thing with markdown is you can use relative links.
replacing that giant URL with `[using credential](samples/sync_samples/client_secrety_auth.py)` will work.

File: sdk/eventhub/azure-eventhub/migration_guide.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+V5 has both sync and async APIs. Sync API is under package azure.eventhub whereas async API is under package azure.eventhub.aio.
+They have the same class names under the two packages. For instance, class `EventHubConsumerClient` with sync API under package `azure.eventhub` has its 
+async counterpart under package `auzre.eventhub.aio`.
+The code samples in this migration guide use async APIs.
+
+### Client constructors
+
+| In v1                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
```
Comment (conniey) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: nit:
`|---|` is enough to for MD to know it's a column. Imo, it's annoying to have to align the headers when you edit any text in a table cell. Markdown is good about rendering these correctly.

File: sdk/eventhub/azure-eventhub/migration_guide.md (hunk 1 above)
Comment (conniey) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: To be consistent, i'd also add back ticks on azure.eventhub and azure.eventhub.aio

File: sdk/eventhub/azure-eventhub/migration_guide.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+Becomes this in V5:
+
+```python
+logger = logging.getLogger("azure.eventhub")
+async def on_event(partition_context, event):
+    logger.info("Message received:{}".format(event.body_as_str()))
+
+client = EventHubConsumerClient.from_connection_string(
+    conn_str=CONNECTION_STR, consumer_group="$default", eventhub_name=EVENTHUB_NAME
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: we should either put `eventhub` param into the `from_connection_string` in T1 or remove the `eventhub_name` param here to keep consistent.

both having eventhub or not:
```python
T1: client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
T2: client = EventHubConsumerClient.from_connection_string(
    conn_str=CONNECTION_STR, consumer_group="$default", eventhub_name=EVENTHUB_NAME
)
```

File: sdk/eventhub/azure-eventhub/migration_guide.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+    await client.run_async()
+    event_data = EventData(b"A single event")
+    await sender.send(event_data)
+finally:
+    await client.stop_async()
+```
+
+In V5:
+```python
+producer = EventHubProducerClient.from_connection_string(conn_str=EVENT_HUB_CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/migration_guide.md]: same comment for the `eventhub`/`eventhub_name`

File: sdk/eventhub/azure-eventhub/migration_guide.md (hunk 2 above)
Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/migration_guide.md]: Good suggestion. Changed

File: sdk/eventhub/azure-eventhub/migration_guide.md (hunk 3 above)
Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/migration_guide.md]: Good suggestion. Changed

File: sdk/eventhub/azure-eventhub/migration_guide.md (hunk 4 above)
Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/migration_guide.md]: Change to use eventhub=EVENTHUB_NAME for v1

File: sdk/eventhub/azure-eventhub/migration_guide.md (hunk 5 above)
Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/migration_guide.md]: Change to use eventhub=EVENTHUB_NAME for v1

**Pull request merged**